src/fhirtypes/StructureDefinition.ts

```diff
@@ -472,7 +472,7 @@ export class StructureDefinition {
     let matchingType: ElementDefinitionType;
     const matchingXElements = elements.filter(e => {
       if (e.path.endsWith('[x]')) {
-        for (const t of e.type) {
+        for (const t of e.type ?? []) {
```

Comment:
Good catch! But I notice that you had to do this here, as well as in another location. I imagine we might have to iterate over `type` in other places in the future, as well.

Is there a reason that we can't just modify the `type` property itself to be set to an empty array if it isn't otherwise set? That might prevent us from having to catch cases like this in the future.

Reply:
Yeah that is a good point. I am not sure how to go about setting `type` though, because we don't want to output JSON with an empty array for `type`.

Reply:
My one concern there is that null types are usually something we inherit from the parent.  So when we serialize back out to JSON, we'd have to make sure that we don't change what _was_ a null type to an empty type ([]) instead.  Might have to put special logic in the serialized to make that happen.

Another possible approach would be to introduce a `nullSafeType` getter or function and try to always use that instead.  But then I guess we need to ask if that's any easier than remembering to do `?? []` as above.

Reply:
Given the issue Chris raises, I would consider rescinding my suggestion. If there's a distinct difference between null and empty for serialization, then getting around that would arguably be more unclear than what we are doing.